Adds new “symbol” config option and deprecates “inner_contents” (#506)
thephpleague/commonmark: docs/1.5/extensions/heading-permalinks.md

This should be a `string` you want prepended to HTML IDs.  This prevents generating HTML ID attributes which might conflict with others in your stylesheet.  A dash separator (`-`) will be added between the prefix and the ID.  You can instead set this to an empty string (`''`) if you don't want a prefix.

### `inner_contents` _(deprecated since `1.5.0`)_

> This controls the HTML you want to appear inside of the generated `<a>` tag.  Usually this would be something you'd style as some kind of link icon, but you can replace this with any custom HTML you wish.
> 
> From versions `1.4.0` to `1.4.3`, the default value for this config option was an embedded [Octicon link SVG](https://iconify.design/icon-sets/octicon/link.html).
> 
> In order to deprecate this config option, its default value had to be removed in version `1.5.0`. While this is
> technically a breaking change, it can be easily restored by setting this config option to the same constant as before
> (note: this constant has also been deprecated):
> 
> ```php
> use League\CommonMark\Extension\HeadingPermalink\HeadingPermalinkRenderer;
> 
> $config = [
>     'heading_permalink' => [
>         'inner_contents' => HeadingPermalinkRenderer::DEFAULT_INNER_CONTENTS,
>     ],
> ];
> ```
> 
> Whenever this config option is provided a value, a deprecation warning will be triggered and the `symbol` config
> option below will be ignored completely.

Reviewer: Assuming we add instructions to the upgrading guide, I think we could simply this to:

```suggestion
This controls the HTML you want to appear inside of the generated `<a>` tag.  Usually this would be something you'd style as some kind of link icon, but you can replace this with any custom HTML you wish.

This option was deprecated in 1.5.0 and will be removed in 2.0.0.  Use the `symbol` option instead.  See the [Upgrade Guide](/1.5/upgrading/) for more information.
```

We'll need to provide information in that upgrade guide too, and based on my other comment I think it should include information on how to keep the Octicon link, either as a CSS rule (recommended) or an embedded SVG (the way it is now, not recommended but still supported)

Also, note that we _might_ end up having a 1.4.4 release if there are any security issues, which is why I removed the reference to 1.4.3. Hopefully not, though! :crossed_fingers:

Author: Done

### `insert`

This controls whether the anchor is added to the beginning of the `<h1>`, `<h2>` etc. tag or to the end.  Can be set to either `'before'` or `'after'`.

### `symbol`

This option sets the symbol used to display the permalink on the document. This defaults to `\League\CommonMark\Extension\HeadingPermalink\HeadingPermalinkRenderer::DEFAULT_SYMBOL = '¶'`.

If you want to use a custom icon, then set this to an empty string `''` and check out the [Adding Icons](#adding-icons) sections below.

> > Note: the symbol should only be is a single character value; additional characters will be stripped (does not affect multibyte characters).
> 

Reviewer: ```suggestion
```

Reviewer: ```suggestion
> Note: Any HTML characters provided here will be escaped for security reasons.

```
